akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerSpec.scala

```diff
@@ -648,6 +648,52 @@ class HttpServerSpec extends AkkaSpec(
       netOut.expectComplete()
     })
 
+    "Don't produce a `100 Continue` response if `Strict` entity is empty because `Content-Length` header is 0" in assertAllStagesStopped(new TestSetup {
+      send("""POST / HTTP/1.1
+             |Host: example.com
+             |Expect: 100-continue
+             |Content-Length: 0
+             |
+             |""")
+
+      val HttpRequest(POST, _, _, entity @ Strict(ContentTypes.NoContentType, ByteString.empty), _) = expectRequest()
+      responses.sendNext(HttpResponse(entity = entity))
```

Comment:
Sending back the request entity here is allowed but it's also a bit of a stretch of the HTTP protocol. We should at least also have a test sending back an arbitrary response that doesn't reference the request entity.